sleuth_backend/views/views.py

```diff
@@ -122,67 +119,89 @@ def search(request):
+
+    return JsonResponse(responses)
 
-def _build_search_query(core, query_str, base_kwargs):
+def getdocument(request):
     '''
-    Builds a serach query and sets parameters that is most likely to
-    return the best results for the given core using the given user query.
-    
-    See https://lucene.apache.org/solr/guide/6_6/the-standard-query-parser.html
-    for more information about Apache Lucene query syntax.
+    Takes GET request containing an ID URL and returns the associated
+    document from Solr database.
+
+    Params:
+        id: the id to search for and retrieve
+        type: (optional) the name of the core to look for the ID in - all cores by default
+        return: (optional) the desired return fields - see sleuth_backend.solr.models
+        state: (optional) any string to be returned in response
+
+    Example Usage:
+    http:// ... /api/get/?id=https://www.ubc.ca&return=links
+
+    Example Response Body:
+    {
+        "data": {
+            "type": <core name>,
+            "doc": { <document format depends on core> }
+        },
+        "request": { <data about your original request> }
+    }
     '''
-    kwargs = base_kwargs
-
-    if core == "genericPage":
-        fields = {
-            'id': 1,
-            'siteName': 10,
-            'name': 10,
-            'description': 5,
-            'content': 2
-        }
-        query = Query(query_str, fields=fields, proximity=5)
-        terms_query = Query(query_str, fields=fields, as_phrase=False)
-        query.select_or(terms_query)
-        kwargs['default_field'] = 'content'
-        kwargs['highlight_fields'] = 'content'
-
-    elif core == "courseItem":
-        fields = {
-            'id': 1,
-            'name': 9,
-            'description': 8,
-            'subjectData': 5,
-        }
-        query = Query(query_str, fields=fields)
-        kwargs['default_field'] = 'name'
-        kwargs['highlight_fields'] = 'description'
+    if request.method != 'GET':
+        return HttpResponse(status=405)
+
+    cores_to_search = build_core_request(request.GET.get('type', ''), SOLR.core_names())
+    doc_id = request.GET.get('id', '')
+    state = request.GET.get('state', '')
+    return_fields = build_return_fields(request.GET.get('return', ''))
+
+    kwargs = { 'return_fields': return_fields }
 
-    else:
-        query = Query(query_str)
+    if doc_id is '':
+        sleuth_error = SleuthError(ErrorTypes.INVALID_GETDOCUMENT_REQUEST)
+        return HttpResponse(sleuth_error.json(), status=400)
+
+    response = {
+        'data': {},
+        'request': {
+            'query': doc_id,
+            'types': cores_to_search,
+            'return_fields': return_fields.split(","),
+            'state': state
+        }
+    }
+    for core_to_search in cores_to_search:
+        try:
+            new_query, new_kwargs = build_getdocument_query(doc_id, kwargs)
+            query_response = SOLR.query(core_to_search, new_query, **new_kwargs)
```

Comment:
I think the `try-except` should just cover this line (line 182) since it's the only thing that throws a SolrError.

Reply:
Are you sure? It also catches things like `KeyError` and `ValueError`

Reply:
Oh my bad. I guess it's fine then. Just try avoid large `try` blocks where you can.